figures/permissions.py

```diff
@@ -72,11 +81,11 @@ class IsSiteAdminUser(BasePermission):
     Would `has_object_permission` help simplify filtering by site?
     """
     def has_permission(self, request, view):
-        return is_site_admin_user(request)
+        return is_site_admin_user(request) or has_insights_access(request)
```

Comment:
Whats the reason to add the insights permission check for site admin ?

Reply:
@burhandodhy this helper function checks if the user belongs to the `edly sub-organization.`